api/src/database/migrations/20240311A-deprecate-webhooks.ts

```diff
@@ -0,0 +1,138 @@
+					collections: toArray(webhook.collections),
+				},
+				operation: null, // Fill this in later --> `operationIdRunScript`
+				// date_created: Default Date,
+				// user_created: null,
+			};
+
+			const operationRunScript = {
+				id: operationIdRunScript,
+				name: 'Transforming Payload for Webhook',
+				key: 'payload-transform',
+				type: 'exec',
+				position_x: 19,
+				position_y: 1,
+				options: {
+					code: '/**\n * Webhook data slightly differs from Flow trigger data.\n * This flow converts the new Flow format into the older Webhook shape\n * in order to not break existing logic of consumers.\n */ \nmodule.exports = async function(data) {\n    const crudOperation = data.$trigger.event.split(".").at(-1);\n    const keyOrKeys = crudOperation === "create" ? "key" : "keys";\n    return {\n        event: `items.${crudOperation}`,\n        accountability: { user: data.$accountability.user, role: data.$accountability.role },\n        payload: data.$trigger.payload,\n        [keyOrKeys]: data.$trigger?.[keyOrKeys],\n        collection: data.$trigger.collection,\n    };\n}',
+				},
+				resolve: operationIdWebhook,
+				reject: null,
+				flow: flowId,
+				// date_created: "",
+				// user_created: "",
+			};
+
+			const operationWebhook = {
+				id: operationIdWebhook,
+				name: 'Webhook',
+				key: 'webhook',
+				type: 'request',
+				position_x: 37,
+				position_y: 1,
+				options: {
+					url: webhook.url,
+					body: webhook.data ? '{{$last}}' : undefined,
+					method: webhook.method,
+					headers: typeof webhook.headers === 'string' ? parseJSON(webhook.headers) : webhook.headers,
+				},
+				resolve: null,
+				reject: null,
+				flow: flowId,
+				// date_created: "",
+				// user_created: "",
+			};
+
+			await trx(TABLE_FLOWS).insert(newFlow);
+
+			// Order is important due to IDs
+			await trx(TABLE_OPERATIONS).insert(operationWebhook);
+			await trx(TABLE_OPERATIONS).insert(operationRunScript);
+
+			await trx(TABLE_FLOWS).update({ operation: operationIdRunScript }).where({ id: flowId });
+
+			// Persist new Flow/s so that we can retroactively disable them on potential down-migrations
+			await trx(TABLE_WEBHOOKS)
+				.update({ [NEW_COLUMN_FLOW]: flowId })
+				.where({ id: webhook.id });
+
+			// 2. Disable existing webhooks
+			await trx(TABLE_WEBHOOKS).update({ status: 'inactive' });
+		}
+	});
+}
+
+/**
+ * Dont completely drop Webhooks yet.
+ * Lets preserve the data and drop them in the next release to be extra safe with users data.
+ */
+export async function down(knex: Knex): Promise<void> {
+	// Set existing migrated flows to inactive
+	// Note: Here we assume that they have not been expanded upon i.e. only disable not delete just in case
+	const migratedFlowIds = (await knex(TABLE_WEBHOOKS).select(NEW_COLUMN_FLOW).whereNotNull(NEW_COLUMN_FLOW)).map(
+		(col) => col[NEW_COLUMN_FLOW],
+	);
+
+	await knex(TABLE_FLOWS).update({ status: 'inactive' }).whereIn('id', migratedFlowIds);
```

Comment:
I think we should drop them instead, otherwise it might get messy when re-doing the migration

Reply:
Alright, done :+1: